streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreBuilderWrapper.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.internals;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+import org.apache.kafka.streams.state.internals.SessionStoreBuilder;
+import org.apache.kafka.streams.state.internals.TimestampedWindowStoreBuilder;
+import org.apache.kafka.streams.state.internals.VersionedKeyValueStoreBuilder;
+import org.apache.kafka.streams.state.internals.WindowStoreBuilder;
+
+public class StoreBuilderWrapper implements StoreFactory {
```

Comment:
Ok one more thing: the javadocs explaining the StoreFactory are great, but I think this needs javaodcs as well. Is it just the default implementation of the StoreFactory class? Is it just here to cover the window/session store cases because they don't yet have a Materializer? etc

Reply:
It's here mostly to cover use cases where the user passes in `StoreBuilder` directly (via PAPI or DSL#processes funs) - so it's here to stay :)

Reply:
Aha -- thanks for clarifying. So IIUC it is like a "wrapper" for already-instantiated StoreBuilders? The current name actually makes total sense once I understood what the class does. Of course ideally the reverse is true...that you can understand what the class does based on the name. But maybe it's obvious and I just lack reading comprehension skills 🤷‍♀️ 

I can't think of a better name, but I feel better now that there's a description for this. Thanks!